src/main/java/com/artgallery/infrastructure/ArtRepositoryDatabase.java

```diff
@@ -0,0 +1,124 @@
+package com.artgallery.infrastructure;
+
+import static com.artgallery.dao.db.Tables.AUTHOR;
+import static com.artgallery.dao.db.Tables.CURATOR;
+import static com.artgallery.dao.db.Tables.PAINTING;
+
+import com.artgallery.dao.db.tables.records.AuthorRecord;
+import com.artgallery.dao.db.tables.records.CuratorRecord;
+import com.artgallery.domain.ArtRepository;
+import com.artgallery.domain.model.Author;
+import com.artgallery.domain.model.Curator;
+import com.artgallery.domain.model.Painting;
+import java.util.List;
+import org.jooq.DSLContext;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ArtRepositoryDatabase implements ArtRepository {
+
+    private final DSLContext dslContext;
+
+    public ArtRepositoryDatabase(DSLContext dslContext) {
+        this.dslContext = dslContext;
+    }
+
+    @Override
+    public List<Painting> getPaintings() {
+        return dslContext.selectFrom(PAINTING)
+            .fetch()
+            .stream()
+            .map(painting -> ArtMapper.mapPainting(painting, findAuthor(painting.getAuthorId()), findCurator(painting.getCuratorId())))
+            .toList();
+    }
+
+    private AuthorRecord findAuthor(Long id) {
+        return dslContext.selectFrom(AUTHOR)
+            .where(AUTHOR.ID.eq(id))
+            .fetchOne();
+    }
+
+    private CuratorRecord findCurator(Long id) {
+        return dslContext.selectFrom(CURATOR)
+            .where(CURATOR.ID.eq(id))
+            .fetchOne();
+    }
+
+    @Override
+    public List<Author> getAuthors() {
+        return dslContext.selectFrom(AUTHOR)
+            .fetch()
+            .map(ArtMapper::mapAuthor);
+    }
+
+    @Override
+    public List<Curator> getCurators() {
+        return dslContext.selectFrom(CURATOR)
+            .fetch()
+            .map(ArtMapper::mapCurator);
+    }
+
+    @Override
+    public void addPainting(Painting painting) {
+        Author author = painting.getAuthor();
+        Long authorId = addAuthorId(author);
+        Curator curator = painting.getCurator();
+        Long curatorId = addCuratorId(curator);
+        dslContext.insertInto(PAINTING,
+                PAINTING.CURATOR_ID, PAINTING.AUTHOR_ID, PAINTING.TITLE, PAINTING.YEAR, PAINTING.MOVEMENT, PAINTING.STATUS)
+            .values(curatorId, authorId, painting.getTitle(), painting.getYear(), painting.getMovement().name(), painting.getStatus().name())
+            .execute();
+    }
+
+    private Long addAuthorId(Author author) {
```

Comment:
addAuthor a nie addAuthorId, to że zwraca ID nie oznacza, że dodałaś tylko ID

Reply:
no ale ta metoda zwraca Long id,
a ja muszę przypisać authora do paintinga, więc zrobiłam drugą metodę i, bo na mnie program krzyczał :C

Reply:
że nie może być 2 takich samych nazw metod :C